### DISCUSSION TITLE
Nuxt 3 Redirecting to an external URL in middleware.

### DISCUSSION THREAD

**Discussion by ggggg** [nuxt/nuxt#17294] [General]
This is my code, but it navigates to localhost:3000/pointlife.net
```ts
import { addRouteMiddleware, defineNuxtPlugin, navigateTo } from '#app'

export default defineNuxtPlugin(() => {
  addRouteMiddleware('auth-test', (to) => {
    console.log(to.meta)
    if (!to.meta.accessLevel || to.meta.accessLevel < 0) {
      return
    }
    console.log('big bad')
    return navigateTo({ path: 'https://pointlife.net', force: true, replace: true })
  }, { global: true })
})
```

**Comment by UnderKoen** [comment]
You can use `window.location` to change to external urls.

This only works on the client so you need to check for this.
```
if (process.client) window.location.href = "https://pointlife.net";
```

**Reply by ggggg** [reply]
This renders the page and sends it to the client, right now my best solution is to navigateTo an empty page and use client redirect (window.location.replace) in beforeMount

**Comment by TheAlexLichter** [comment]
IMO it'd be nice to have that functionality in `navigateTo`. I've created [a PR](https://github.com/nuxt/framework/pull/5022) to add naive functionality handling external links.

**Comment by justin-schroeder** [comment]
Posted this on #5022 as well, but until that is merged here's a little composable function you can use to achieve an external redirect: https://gist.github.com/justin-schroeder/f583797c702155f32c9078d1add2a594
